DOC-6250 Eventing (replacement) High Risk example

jon-strabala opened the pull request:
New complete Eventing High Risk example with needed JSON data files.

DOC-6250 (also DOC-4176 5.5 and DOC-4428 5.5)

Since there was NO data files available to the end user this was completely redone.

jon-strabala committed:
Update eventing-examples-high-risk.adoc

jon-strabala committed:
Add files via upload

jon-strabala committed:
Create _place_holder_

jon-strabala committed:
needed data files for new high risk example

jon-strabala committed:
remove item needed to make directory

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+Linux::
++
+--
+Assuming of the downloaded files needed are in /tmp (note your username and password may differ)
```
```suggestion
Assuming that the downloaded files needed are in /tmp (note your username and password may differ)
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+macOS::
++
+--
+Assuming of the downloaded files needed are in /Users/$USER/Downloads (note your username and password may differ)
```
```suggestion
Assuming that the downloaded files needed are in /Users/$USER/Downloads (note your username and password may differ)
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+Windows::
++ 
+-- 
+Assuming of the downloaded files needed are in "C:\Users\%USERNAME%\Downloads" (note your username and password may differ)
```
```suggestion
Assuming that the downloaded files needed are in "C:\Users\%USERNAME%\Downloads" (note your username and password may differ)
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+The resulting dialog indicates that a purchase of an iMac was flagged as the credit card's default currency was USD, but the purchase was in GBP, e.g. made with a merchant in a foreign country using a foreign currency.
++
+[source,console]
```
```suggestion
[source,json]
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+An example record from the _cards.json_ file that you just downloaded encapsulates the information of a credit card:
++
+[source,console]
```
```suggestion
[source,json]
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+An example record from the _merchants.json_ file that you just downloaded encapsulates the information of a merchant:
++
+[source,console]
```
```suggestion
[source,json]
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+An example record from the _exchange_rates.json_ file that you just downloaded encapsulates the information of a set of exchange rates:
++
+[source,console]
```
```suggestion
[source,json]
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+
+. An example record from the _txns.json_ file that you just downloaded encapsulates the information of a transaction or a card charge:
++
+[source,console]
```
```suggestion
[source,json]
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
 
 *Goal*: This example illustrates how to leverage Eventing Service in the Banking and Financial domain.
-When a credit card transaction exceeds the user’s available credit limit, to indicate a high-risk transaction, an alert can be generated.
+When a credit card transaction either 1) exceeds the user’s available credit limit or 2) a transaction in a foreign currency is made -  a high-risk transaction, an alert can be generated.
```
```suggestion
When a credit card transaction either 1) exceeds the user’s available credit limit or 2) is made in a foreign currency, a high-risk transaction alert can be generated.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+
+. Lookup details from the customers card information specifically: date, spending limits, location, and default currency. This data is from bucket *'register'* with type='card'.
+. Lookup the exchange rates for the date from the transaction. This data is from bucket *'register'* with type='exchange_rates'.
+. Normalize the currencies from both the transaction and the spending limit into USD (at the correct days exchange rate).
```
```suggestion
. Normalize the currencies from both the transaction and the spending limit into USD (at the correct day's exchange rate).
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+. Determine if the purchase is a foreign purchase (this is high risk).
+. For all high-risk transaction generate a new document containing both some transaction information and some calculated data.
+. Write the new document representing the suspect transaction to the *'review'* bucket with type='transaction'.
+. [OPTIONAL] Use of cURL directly from Eventing, use of a custom application written in various Couchbase SDKs, or third-party integrations like Kafka can then read the items written to the *'review'* bucket (type='transaction') for further automated action.
```
```suggestion
. [OPTIONAL] Use of cURL directly from Eventing, use of a custom application written in various Couchbase SDKs, or use of third-party integrations like Kafka to read the items written to the *'review'* bucket (type='transaction') for further automated action.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
 
 For steps to create buckets, see xref:manage:manage-buckets/create-bucket.adoc[Create a Bucket].
 
+NOTE: The *'metadata'* bucket is for the sole use of the Eventing system, do not add, modify, or delete documents from this bucket. In addition, do not drop or flush or delete the bucket while you have any deployed Eventing functions.
```
```suggestion
NOTE: The *'metadata'* bucket is for the sole use of the Eventing system. Do not add, modify, or delete documents from this bucket. In addition, do not drop or flush or delete the bucket while you have any deployed Eventing functions.
```

jon-strabala replied:
This is boiler plate in the other Examples (as such the same update should be done).

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+
 *Procedure*:
 
+. Modify the *'review'* bucket to enable the ability to "flush" the items from the UI (_do not do this for the other two buckets_). Access the *Couchbase Web Console* > *Buckets* page, select the bucket '*review*' to expand the buckets view.
```
```suggestion
. Modify the *'review'* bucket to enable the ability to "flush" the items from the UI (_do not do this for the other two buckets_). Access the *Couchbase Web Console* > *Buckets* page, and select the bucket '*review*' to expand the bucket view.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+CREATE INDEX `adv_type` ON `review`(`type`);
+----
++
+Copy and paste the two lines of text (above) into the Query Workbench and click *'Execute'*
```
```suggestion
Copy and paste the two lines of text (above) into the Query Workbench and click *'Execute'*.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+image::high_risk_txns_05_create_indexes.png[,100%]
++
+Although the above indices are technically not needed for our Eventing function they will come in useful for exploring
+the documents imported to the *'register'* bucket and inserted into the '*review*' by our Eventing Function using the N1QL langague in the Query Workbench 
```
```suggestion
Although the above indexes are technically not needed for our Eventing function they will come in useful for exploring
the documents imported to the *'register'* bucket and inserted into the '*review*' by our Eventing Function using the N1QL language in the Query Workbench.
```

Our style guide prefers indexes, not indices

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
-** After configuring your settings your screen should look like:
+  *** For the first binding, select "bucket alias", specify *register* as the "alias name" of the bucket, and select *register* as the associated bucket, and select "read only".
+  *** For the second binding, select "bucket alias", specify *review* as the "alias name" of the bucket, and select *review* as the associated bucket, and select "read and write".  
+ ** After configuring your settings your screen should look like:
```
```suggestion
 ** After configuring your settings your dialog should look like this:
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
-image::highrisk_03_editor_with_code.png[,100%]
+image::high_risk_txns_03_code.png[,100%]
 ** Click *Save*.
 ** To return to the Eventing screen, click the '*< back to Eventing*' link (below the editor) or click *Eventing* tab.
```
```suggestion
** To return to the Eventing screen, click the '*< back to Eventing*' link (below the editor) or click the *Eventing* tab.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
-If the transaction currency is other than the preconfigured home currency of the user, then the handler flags the transactions and moves it to a different bucket.
+In addition, the handler checks if a foreign currency purchase has occurred, this is also flagged as a high-risk transaction.
++
+The Function *high_risk_txns* then copies the transaction to a different bucket to the *review* bucket (but it removed some unneeded data and also adds some enriched data). The handler enriches the document with predefined _comments_ and also provides a _reason code_. In the last part, the handler performs a currency validation step.
```
```suggestion
The Function *high_risk_txns* then copies the transaction to the *review* bucket (but it removes some unneeded data and adds some enriched data). The handler enriches the document with predefined _comments_ and also provides a _reason code_. In the last part, the handler performs a currency validation step.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+The handler also converts both the credit limit and the transaction amount to a common currency, in this case USD, based upon current exchange rates on the exact date of the given transaction.
+
+. Now we will seed the required sample data, there are a total of four (4) data files that need to be downloaded to your Couchbase instance via a "_Right click, Save Link As ...._" for remote instances "_Right click, Copy Link Address_" and use either the cURL or wget utility to download the files:
```
```suggestion
. Now we will seed the required sample data. There are a total of four (4) data files that need to be downloaded to your Couchbase instance. Right-click on each of the links below and choose *Save Link As* to download the files. For remote instances, right-click on each of the links below and choose *Copy Link Address*, then use either the cURL or wget utility to download the files.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+| Credit card information
+| type='card'
+| 7
+| link:./_attachments/examples/high_risk/cards.json[Download,window=_blank]
```
```suggestion
| link:{attachmentsdir}/examples/high_risk/cards.json[Download,window=_blank]
```
Let Antora handle the path to the attachments.

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+| Merchant information
+| type='merchant'
+| 5001
+| link:./_attachments/examples/high_risk/merchants.json[Download,window=_blank]
```
```suggestion
| link:{attachmentsdir}/examples/high_risk/merchants.json[Download,window=_blank]
```
Let Antora handle the path to the attachments.

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+| Daily exchange rates
+| type='exchange_rates'
+| 422
+| link:./_attachments/examples/high_risk/exchange_rates.json[Download,window=_blank]
```
```suggestion
| link:{attachmentsdir}/examples/high_risk/exchange_rates.json[Download,window=_blank]
```
Let Antora handle the path to the attachments.

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+| Credit Card charges
+| type='transaction'
+| 417
+| link:./_attachments/examples/high_risk/txns.json[Download,window=_blank]
```
```suggestion
| link:{attachmentsdir}/examples/high_risk/txns.json[Download,window=_blank]
```
Let Antora handle the path to the attachments.

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+}
+----
+
+. The downloaded files now need to all be loaded into the *register* bucket, this can be done as follows:
```
```suggestion
. The downloaded files now need to all be loaded into the *register* bucket. This can be done as follows:
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+--
+====
+
+. Access the *Couchbase Web Console* > *Buckets* page, notice that a total of 5,847 documents have been loaded into the bucket '*register*'.
```
```suggestion
. Access the *Couchbase Web Console* > *Buckets* page and notice that a total of 5,847 documents have been loaded into the bucket '*register*'.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+image::high_risk_txns_06_json_loaded.png[,100%]
+
+. We are now ready to start the Eventing function. From the *Couchbase Web Console* > *Eventing* screen:
+** click on the Function name *high_risk_txns* to expand and expose the Function controls.
```
```suggestion
** Click on the Function name *high_risk_txns* to expand and expose the Function controls.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+image::high_risk_txns_04_controls.png[,100%]
++
+** click *Deploy*.
 ** In the *Confirm Deploy Function* dialog, select *Everything from the Feed boundary* option.
```
```suggestion
** Click *Deploy*.
** In the *Confirm Deploy Function* dialog, select *Everything* from the *Feed boundary* option.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+image::high_risk_txns_07_log.png[,100%]
++
+** the dialog should have data be similar to the following (only a few selected lines are displayed below):
```
```suggestion
** The dialog should have data similar to the following (only a few selected lines are displayed below):
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+image::high_risk_txns_08_bucket_documents.png[,%100]
++
+** You can select any document in the *review* bucket, in addition you can select by a specific KEY for example 'transaction:tx-1505315650-403' by cutting and pasting this ID (without quotes) into the *'Document ID'* text box and hitting a return
```
```suggestion
** You can select any document in the *review* bucket. In addition you can select by a specific key, for example `transaction:tx-1505315650-403`, by cutting and pasting this ID into the *Document ID* text box and hitting kbd:[Return].
```
Better to use formatting, rather than quotes, to delimit the text to be entered. This means you don't need to remind users not to enter the quotes.

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
-. From the *Couchbase Web Console* > *Query* page, execute the below N1QL query:
+. From the *Couchbase Web Console* > *Query* page, run a few N1QL queries on the new documents create in the *'review*' bucket:
++
+*QUERY A*, see how many high-risk transactions we found
```
```suggestion
*QUERY A*, see how many high-risk transactions we found.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+image::high_risk_txns_09_n1ql_a.png[,%100]
++
+*QUERY B*, look at all the data and with a specific order
```
```suggestion
*QUERY B*, look at all the data and with a specific order.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+image::high_risk_txns_09_n1ql_b.png[,%100]
++
+*QUERY C*, look at the summarized data and with a specific order and grouping
```
```suggestion
*QUERY C*, look at the summarized data and with a specific order and grouping.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+image::high_risk_txns_09_n1ql_f.png[,%100]
+
+. The next step is to follow the Function logic in detail, to do this 'flush' all the generated documents in review, access 
+the *Couchbase Web Console* > *Buckets* page. You should see 40 documents in the '*review*' bucket.  Select the bucket '*review*' to expand the buckets view. Next click on the *'Flush'* button in a while the document count will drop to 0.
```
```suggestion
. The next step is to follow the Function logic in detail. To do this we need to 'flush' all the generated documents in review.
Access the *Couchbase Web Console* > *Buckets* page. You should see 40 documents in the '*review*' bucket.  Select the bucket '*review*' to expand the bucket view. Next click on the *'Flush'* button.
+
** In the *Confirm Flush Bucket* dialog, click *Flush Bucket*. In a while the document count will drop to 0.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+image::high_risk_txns_10_flush_review.png[,%100]
+
+. From the *Couchbase Web Console* > *Eventing* page, click *Pause*.
```
```suggestion
. From the *Couchbase Web Console* > *Eventing* page, click *high_risk_txns* to expand the function, and then click *Pause*.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+image::high_risk_txns_10_pause_confirm.png[,344]
++
+** The Eventing function is paused in a few seconds and can be edited. Click the *'Edit JavaScript'* button.
+** In the Editor dialog change the OnUpdate handler code from *'var verbose = 0'* to *'var verbose = 3'* you are only modifying line four of the *high_risk_txrns" Function as below:
```
```suggestion
** In the Editor dialog change the OnUpdate handler code from `var verbose = 0` to `var verbose = 3` -- you are only modifying line four of the *high_risk_txns* function as below:
```
Again, use formatting rather than quotes, so the users do not enter the quotes.

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+    // *** many lines not shown ***
+----
+** Click *'Save'*.
+** To return to the Eventing screen, click the *'< back to Eventing'* link (below the editor) or click *Eventing* tab.
```
```suggestion
** To return to the Eventing screen, click the *'< back to Eventing'* link (below the editor) or click the *Eventing* tab.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+. From the *Eventing* screen, click *Resume*.
++
+image::high_risk_txns_10_resume.png[,%100]
+** In the *Confirm Resume Function* dialog, Click *Resume Function*.
```
```suggestion
** In the *Confirm Resume Function* dialog, click *Resume Function*.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+image::high_risk_txns_10_resume_confirm.png[,346]
+
+. The Eventing function is resumed the check point created when you 'Paused' the function and will start running within a few seconds. The defined Function is executed on all new document documents and on subsequent mutations. Until a mutation is triggered there will be no processing at all by our modified Handler's JavaScript code.
```
```suggestion
. The Eventing function is resumed from the check point created when you paused the function and will start running within a few seconds. The defined function is executed on all new document documents and on subsequent mutations. Until a mutation is triggered there will be no processing at all by our modified handler's JavaScript code.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+. The next step is to create one mutation, to do this access 
+the *Couchbase Web Console* > *Buckets* page. 
+** Click the *Documents* link of the *register* bucket (the source of the mutations that our Function watches). 
+** Select the document 'transaction:tx-1505315650-403' by cutting and pasting this ID (without quotes) into the *'Document ID'* text box and hitting a return.  
```
```suggestion
** Select the document `transaction:tx-1505315650-403` by cutting and pasting this ID into the *Document ID* text box and hitting kbd:[Return].  
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+image::high_risk_txns_10_bucket_register_id.png[,%100]
++
+** This should display a single document, edit it by Clicking on it.
+** Adjust the document slightly from '"amount": 5383.35' to '"amount": 5383.36'
```
```suggestion
** This should display a single document. Edit it by clicking on it.
** Adjust the document slightly from `"amount": 5383.35` to `"amount": 5383.36`.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+image::high_risk_txns_10_modify.png[,580]
++
+** Click *'Save'*
+** Inspect the new data written to the "high_risk_txns.log" in the file system (or alternatively use the UI's Log link)
```
```suggestion
** Click *'Save'*.
** Inspect the new data written to the "high_risk_txns.log" in the file system (or alternatively use the UI's Log link).
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
 +
-image::highrisk_05_n1ql_query.png[,%100]
+Above we see that for one mutation the Function *high_risk_txns* printed the following log messages:
+** The transaction document 'doc' that mutated (what you just changed)
```
```suggestion
** The transaction document 'doc' that mutated (what you just changed).
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+** The 'card' or credit card that the transaction was made with was looked up via the key _'card:' + doc.card_ from the bucket alias  register.
+** A daily 'rates' or exchange rates for the date of the transaction which was looked up via the key _'exchange_rates:er-' + (doc.date).substr(0, 10))_ from the bucket alias register.
+** Next a bit of information (4 lines) was emitted  to show what was done in order to make the final decision.
+** Finally, the last log line is the decision line if the transaction is considered 'High Risk'.
```
```suggestion
** Finally, the last log line is the decision line (if the transaction is considered 'High Risk').
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+. From the *Eventing* screen, click *Undeploy*.
++
+image::high_risk_txns_10_pause.png[,%100]
+** In the *Confirm Undeploy Function* dialog, Click *Undeploy Function*.
```
```suggestion
** In the *Confirm Undeploy Function* dialog, click *high_risk_txns* to expand the function, and then click *Undeploy Function*.
```

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
++
+image::high_risk_txns_11_undeploy_confirm.png[,346]
+
+. The Eventing function will be undeployed within a few seconds, this concludes the Example.
```
```suggestion
. The Eventing function will be undeployed within a few seconds. This concludes the Example.
```

jon-strabala replied:
See next comment

simon-dew commented on modules/eventing/pages/eventing-examples-high-risk.adoc:
```diff
@@ -2,129 +2,671 @@
+
+. The Eventing function will be undeployed within a few seconds, this concludes the Example.
+
+. Cleanup, go to the Eventing portion of the UI and Undeploy the Function *high_risk_txns*, this will remove the 1024 documents (64 for macOS) from the 'metadata' bucket (in the Bucket view of the UI). Remember you may only delete the 'metadata' bucket if there are no deployed Eventing functions. Delete the *'register'* and *'review'* buckets and also the empty *'metadata*' bucket.
```
```suggestion
. Undeploying the function *high_risk_txns* removes the 1024 documents (64 for macOS) from the *metadata* bucket, as you can see in the Bucket view of the UI. Remember you may only delete the *metadata* bucket if there are no deployed Eventing functions. To clean up, delete the *register* and *review* buckets and also the empty *metadata* bucket.
```

jon-strabala replied:
A prefer to something starting  with the word 'Cleanup' it is consistent with the final steps in my other Examples.  I have a reason for the use of the word 'Cleanup' here as I plan to reference this in blog and stress that the final step 'Cleanup' be skipped.

===== maybe the Example can end with the following ====

. Cleanup involves undeploying and deleting the function and then removing the three buckets you created. This concludes the Example.

** From the *Eventing* screen, click *Undeploy*.
+
image::high_risk_txns_10_pause.png[,%100]
+
** In the *Confirm Undeploy Function* dialog, Click *Undeploy Function*.
+
image::high_risk_txns_11_undeploy_confirm.png[,346]

** The Eventing function will be undeployed within a few seconds. 

** Undeploying the function *high_risk_txns* removes the 1024 documents (64 for macOS) from the *metadata* bucket, as can be seen in the Bucket view of the UI. Remember you may only delete the *metadata* bucket if there are no currently deployed Eventing functions.

** To remove all of the artifacts: 1) delete the *high_risk_txns* function 2) delete the three buckets *register*, *review*, and *metadata* (the later *metadata* should be empty).

simon-dew reviewed (changes requested):
Fantastic worked example. Sorry there are so many suggested corrections, but the page has been entirely rewritten and most of my suggestions are minor.

- [x] Note that the attachments `cards.json`, `exchange_rates.json`, `merchants.json`, and `txns.json` are saved in an incorrectly-named directory, so the download links will not work. The directory for these files should be called `attachments` not `attachements`.

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

SImon's work was almost perfect will be making some minor changes after this commit

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala committed:
Update modules/eventing/pages/eventing-examples-high-risk.adoc

Co-Authored-By: Simon Dew <39966290+simon-dew@users.noreply.github.com>

jon-strabala commented:
Thank's Simon, (from your slack directions) I goofed a bit I initially only committed the your last item.

Then I couldn't seem to do them in batch (the "Add suggestion to batch" was always disabled and I couldn't do it the ).  Sort of painful committing them one at a time, and of course it didn't seem right (I did some procedure wrong or perhaps I don't have the correct privileges).  

So in the pull/1145 everything at the top says "outdate" and everything below says "verified"

And now all your updates to my work seem to be in the couchbase 'repo'

jon-strabala committed:
Added my updated cleanup to Simon's work

jon-strabala committed:
correct name attachments

jon-strabala committed:
Rename modules/eventing/assets/attachements/examples/high_risk/exchange_rates.json to modules/eventing/assets/attachments/examples/high_risk/exchange_rates.json

jon-strabala committed:
Rename modules/eventing/assets/attachements/examples/high_risk/txns.json to modules/eventing/assets/attachments/examples/high_risk/txns.json

jon-strabala committed:
delete and re-add tobig for a rename

jon-strabala committed:
remove file and thus bad directory

simon-dew reviewed (approved):
LGTM 👍
